core/iwasm/common/wasm_runtime_common.c

```diff
@@ -3362,6 +3599,24 @@ wasm_runtime_invoke_native(WASMExecEnv *exec_env, void *func_ptr,
                 PUT_F64_TO_ADDR(
                     argv_ret, invokeNative_Float64(func_ptr, argv1, n_stacks));
                 break;
+#if WASM_ENABLE_REF_TYPES != 0
+            case VALUE_TYPE_EXTERNREF:
+            {
+                int32 ret = invokeNative_Int64(func_ptr, argv1, n_stacks);
```

Comment:
Seems should distinguish AOT func or native func, if AOT, the return value is externref index:
```C
int32 ret = invokeNative_Int32(...);
argv_ret[0] = ret;
```
if native func, the return value should be the object pointer:
```C
int64 ret = invokeNative_Int64(...);    => but not int32 ret
convert ret to externref_idx;
argv[0] = externref_idx;
```